### PR TITLE
Rebase 1.12.2

### PR DESCRIPTION
**Pull request by soltysh**

### CODE REVIEW

**Comment by soltysh** [comment]
@bparees and @openshift/sig-developer-experience heads up this bumps docker, also ptal:
- `Extend request timeout for image import tests`


@enj please double check: 
- `UPSTREAM: 70019: Refactor dry run test to reuse ETCD storage data`
- `UPSTREAM: 70630: Add CRDs to etcd storage path test`

@openshift/sig-auth ptal at:
- `UPSTREAM: 70314: Revert "limit forbidden error to details of what was forbidden"`
- `RBAC policy test changes`
- `Update allowed API groups and bootstrap policy` wrt policy updates

@squeed ptal at:
- `Network controller wiring`

**Comment by bparees** [comment]
@soltysh I don't think we're using docker for much(anything?) any more in origin from a devex perspective.

timeout bump seems fine... i mean, slightly disconcerting, but not sufficient to keep me up.

**Comment by soltysh** [comment]
> timeout bump seems fine... i mean, slightly disconcerting, but not sufficient to keep me up.

it might also be my network, but then again docker's mysql repo is growing so it might be that too.

**Comment by squeed** [comment]
Networking changes seem reasonable, but @danwinship should give it a quick check.

File: glide.yaml [hunk 1]
```diff
@@ -65,32 +68,25 @@ import:
   repo:    https://github.com/openshift/containers-image.git
   version: openshift-3.8
 # pod - sjenning
-# origin-3.11-runc-871ba2e
 - package: github.com/opencontainers/runc
-  repo:    https://github.com/openshift/opencontainers-runc.git
```
Comment (sttts) [inline glide.yaml]: @soltysh all these removals of the repo lines are intentional?

Comment (soltysh) [reply glide.yaml]: Yes, it's expected and intentional. I've made my due diligence before doing so.

File: test/integration/template_test.go [hunk 2]
```diff
@@ -84,7 +84,7 @@ func TestTemplate(t *testing.T) {
 			t.Fatalf("unexpected object: %#v", svc)
 		}
 		// clear namespace
-		if meta["namespace"] != "" {
+		if _, ok := meta["namespace"]; ok {
```
[the remark this answers is not in the document]
Comment (soltysh) [reply test/integration/template_test.go]: I'm not 100% sure, but generally the string you're getting back is not an empty one, but some sort of undefined. That's why I needed to swtich to verifying ok.

File: glide.yaml [hunk 3]
```diff
@@ -45,7 +45,8 @@ import:
 - package: github.com/openshift/client-go
   version: master
 - package: github.com/openshift/library-go
-  version: master
+  repo:    https://github.com/soltysh/library-go.git
+  version: rebase-1.12.2
```
Comment (deads2k) [inline glide.yaml]: why did this change?

**Comment by danwinship** [comment]
> Networking changes seem reasonable, but @danwinship should give it a quick check.

lgtm too

File: glide.yaml (hunk 3 above)
Comment (soltysh) [reply glide.yaml]: That's part of my drop commit, that will be removed once we merge https://github.com/openshift/library-go/pull/78

File: glide.yaml (hunk 1 above)
Comment (soltysh) [reply glide.yaml]: I take that back, this change is not necessary, removing it.

File: glide.yaml (hunk 3 above)
Comment (soltysh) [reply glide.yaml]: This is already dropped.

File: pkg/oc/cli/registry/login/login.go [hunk 4]
```diff
@@ -174,10 +175,10 @@ func (o *LoginOptions) Complete(f kcmdutil.Factory, args []string) error {
 			return fmt.Errorf("the service account %s had no valid secrets associated with it", o.ServiceAccount)
 		}
 	default:
-		if len(cfg.BearerToken) == 0 {
+		if cfg.WrapTransport == nil {
```
Comment (soltysh) [inline pkg/oc/cli/registry/login/login.go]: @smarterclayton ptal at this change, this was what was causing the problems. https://github.com/kubernetes/kubernetes/pull/67359 introduced a periodical reading of token file for inClusterConfig and introduced this field.

**Comment by gnufied** [comment]
@soltysh if you are doing this on your branch - please just make sure you pick second commit from https://github.com/openshift/origin/pull/21727/commits PR. The first commit is already included in 1.12 .

**Comment by soltysh** [comment]
> @soltysh I have a branch over here - https://github.com/gnufied/origin/commits/rebase-1.12.2 which resolves the conflicts. I was thinking of opening a PR against your PR but since I rebased against master again, my commit diff was quite big (although code diff was small).

Appreciate your help, but resolving conflicts is not sufficient. I need to ensure we properly bump k8s and re-apply those patches. But I'll definitely use your branch as a reference :)

**Comment by soltysh** [comment]
> @soltysh if you are doing this on your branch - please just make sure you pick second commit from https://github.com/openshift/origin/pull/21727/commits PR. The first commit is already included in 1.12 .

That's a very handy advice :) Thanks!

**Comment by soltysh** [comment]
Superseded by #21776.

**Pull request closed without merging**